src/vivarium_public_health/disease/state.py

```diff
@@ -458,26 +461,56 @@ def setup(self, builder: Builder) -> None:
         super().setup(builder)
         self.clock = builder.time.clock()
 
-        self.dwell_time = self.get_dwell_time_pipeline(builder)
-        self.disability_weight = self.get_disability_weight_pipeline(builder)
+        builder.value.register_attribute_producer(
+            f"{self.state_id}.dwell_time",
+            source=self.lookup_tables["dwell_time"],
+            component=self,
+            required_resources=get_lookup_columns([self.lookup_tables["dwell_time"]]),
+        )
+        builder.value.register_attribute_producer(
+            f"{self.state_id}.disability_weight",
+            source=self.compute_disability_weight,
+            component=self,
+            required_resources=get_lookup_columns([self.lookup_tables["disability_weight"]])
+            + ["alive", self.model],
+        )
 
         builder.value.register_attribute_modifier(
             "all_causes.disability_weight",
-            modifier=self.disability_weight,
+            modifier=lambda index: self.population_view.get_attributes(
+                index=index, attributes=self.dw_pipeline_name
+            ),
             component=self,
         )
 
         self.has_excess_mortality = is_non_zero(
             self.lookup_tables["excess_mortality_rate"].data
         )
-        self.joint_paf = self.get_joint_paf(builder)
-        self.excess_mortality_rate = self.get_excess_mortality_rate_pipeline(builder)
+        paf = builder.lookup.build_table(0)
+        builder.value.register_attribute_producer(
```

Comment:
The reason these methods were broken out of setup is so that if a subclass needed to change how they generated a single one of these pipelines, they wouldn't have to copy the whole setup method and replace it. I don't think that need has changed. I'd suggest retaining the methods as they were, but rename them from `get_x_pipeline()` to `register_x_pipeline()` and have them return None.